scripts/error-handler.sh

```
@@ -0,0 +1,234 @@
+            "$error_type"
+        return 1
+    fi
+    return 0
+}
+
+# Function to check disk space with error handling
+check_disk_space() {
+    local required_mb="$1"
+    local path="${2:-/}"
+    
+    local available_kb=$(df "$path" | awk 'NR==2 {print $4}')
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)